assets/js/components/notifications/ErrorNotifications.test.js

```diff
@@ -45,6 +50,74 @@ describe( 'ErrorNotifications', () => {
 		registry.dispatch( CORE_USER ).receiveGetDismissedItems( [] );
 	} );
 
+	it( 'does not render UnsatisfiedScopesAlert when user is not authenticated', () => {
+		provideUserAuthentication( registry, {
+			authenticated: false,
+			unsatisfiedScopes: [
+				'https://www.googleapis.com/auth/analytics.readonly',
+			],
+		} );
+		provideModules( registry, [
+			{
+				slug: 'analytics-4',
+				active: true,
+				connected: true,
+			},
+		] );
+		provideNotifications(
+			registry,
+			{
+				'authentication-error':
+					DEFAULT_NOTIFICATIONS[ 'authentication-error' ],
+			},
+			true
+		);
+		const { container } = render( <ErrorNotifications />, {
+			registry,
+			viewContext: VIEW_CONTEXT_MAIN_DASHBOARD,
+		} );
+		expect( container.childElementCount ).toBe( 0 );
+	} );
+
+	it( 'renders UnsatisfiedScopesAlert when user is authenticated', async () => {
+		act( () => {
+			provideUserAuthentication( registry, {
+				grantedScopes: [ TAGMANAGER_READ_SCOPE ],
+				unsatisfiedScopes: [
+					'https://www.googleapis.com/auth/analytics.readonly',
+				],
+			} );
+			provideModules( registry, [
+				{
+					slug: 'analytics-4',
+					active: true,
+					connected: true,
+				},
+			] );
+			provideNotifications(
+				registry,
+				{
+					'authentication-error':
+						DEFAULT_NOTIFICATIONS[ 'authentication-error' ],
+				},
+				true
+			);
+		} );
+
+		const { container, waitForRegistry } = render( <ErrorNotifications />, {
+			registry,
+			viewContext: VIEW_CONTEXT_MAIN_DASHBOARD,
+		} );
+
+		await act( waitForRegistry );
+		await act( waitForDefaultTimeouts );
+
+		expect( container ).toHaveTextContent(
+			'Site Kit can’t access necessary data'
+		);
+		expect( container ).toMatchSnapshot();
```

Comment:
Does this need to be a full snapshot? Is there any reason not to test for elements/text content instead?

I just notice there are a lot of snapshot tests in this PR, and they're always somewhat fragile since they're rendering `Button`/`Link` components that can change and cause "unrelated" snapshot changes in tests like this.

Reply:
@tofumatt The ACs here were to bring back all the tests and the original coverage. So I've kept snapshots where ever they were there before and only updated them where necessary.

Reply:
Fair enough! 🙂